### PR TITLE
Update crusty old pinned Rust dependencies, multiple holochain-serialization versions in Cargo.lock

### PR DESCRIPTION
**Pull request by pjkundert**
There are a bunch of brittle downstream dependencies that are pulling in old Rust packages.

The most urgent, https://github.com/freesig/observability.git, is currently fixed to an old version of holochain-serialization.  This must be fixed, as the current `develop` branch of holochain/holochain includes multiple versions of holochain-serialization in its Cargo.lock.

Updating each of freesig/observability, holochain/holochain-serialization and holochain/holochain-wasmer according to their `update-sym` branches will tidy up a lot of the ancient "syn", ... stuff.

Remember to remove the root Cargo.toml's [patch.crates-io] redirects before merging this.  Also, get the test WASM build working...

### CODE REVIEW

**Comment by pjkundert** [comment]
OK, I've rebased this against develop, awaiting merges of downstream dependencies.

Until (at least) the fixes to freesig/observability are merged, holochain/holochain isn't correct, due to multiple differing holochain-serialization inclusions.

The updates to holochain-wasmer and holochain-serialization remove a big set of multiple "syn", "proc-macro2", ... inclusions of ancient versions of these crates, and should also be done.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+SHELL		= /bin/bash
+
+.PHONY: all test
+all:		test
+
+# test -- build required WASM and execute unit tests
+test:
+	cargo build --features 'build_wasms' --manifest-path=crates/holochain/Cargo.toml
+	cargo test
```
Comment (freesig) [inline Makefile:9]: This is already a function in the nix-shell under `hc-merge-test` or `hc-test` I really don't think we need a third way to build holochain?

File: Cargo.toml [hunk 2]
```diff
@@ -51,3 +51,15 @@ lmdb-rkv = { git = "https://github.com/holochain/lmdb-rs.git" }
...
+#holochain_json_api                = { git = "https://github.com/pjkundert/holochain-serialization.git", branch = "update-syn" }
+#holochain_json_derive             = { git = "https://github.com/pjkundert/holochain-serialization.git", branch = "update-syn" }
+holochain_serialized_bytes        = { git = "https://github.com/pjkundert/holochain-serialization.git", branch = "update-syn" }
+holochain_serialized_bytes_derive = { git = "https://github.com/pjkundert/holochain-serialization.git", branch = "update-syn" }
+
+holochain_wasmer_guest            = { git = "https://github.com/pjkundert/holochain-wasmer.git",        branch = "update-syn" }
+holochain_wasmer_common           = { git = "https://github.com/pjkundert/holochain-wasmer.git",        branch = "update-syn" }
+holochain_wasmer_host             = { git = "https://github.com/pjkundert/holochain-wasmer.git",        branch = "update-syn" }
+
+observability                     = { git = "https://github.com/pjkundert/observability.git",           branch = "update-syn" }
```
Comment (freesig) [inline Cargo.toml:65]: I've published this so you should be able to point at 0.1.1 now

File: crates/hdk/src/error.rs [hunk 3]
```diff
@@ -21,6 +22,9 @@ pub enum HdkError {
 
     #[error("A remote zome call was made but there was a network error: {0}")]
     ZomeCallNetworkError(String),
+
+    #[error(transparent)]
+    Infallible(#[from] Infallible),
```
Comment (freesig) [inline crates/hdk/src/error.rs:27]: Any reason for this in this PR? Seems unrelated?

File: crates/hdk/src/guest_callback/entry_defs.rs [hunk 4]
```diff
@@ -66,7 +66,7 @@ macro_rules! entry_def {
             fn try_from(entry: &$crate::prelude::Entry) -> Result<Self, Self::Error> {
                 match entry {
                     $crate::prelude::Entry::App(eb) => Ok(Self::try_from(
-                        $crate::prelude::SerializedBytes::from(eb.to_owned()),
+                        $crate::prelude::SerializedBytes::try_from(eb.to_owned())?,
```
Comment (freesig) [inline crates/hdk/src/guest_callback/entry_defs.rs:69]: Was this a change due to a newer version of serialized bytes?
I don't think you need the `to_owned()`?

**Review by freesig — CHANGES_REQUESTED** [review]
I like the idea of bumping the dependencies and appreciate the effort of making this and the related PRs.
A few things:
- Looks like a lot of `Cargo.lock` files have been deleted. I don't think we want that.
- There's some changes to the HDK that seem unrelated or maybe just need some explanation.

**Comment by freesig** [comment]
I'm a little confused as to why observability would cause conflicts with holochain_serialization as it's under a feature flag that is disabled by default. If that's not working it would be good too know

File: crates/hdk/src/error.rs (hunk 3 above)
Comment (pjkundert) [reply crates/hdk/src/error.rs:27]: This is also due to updating syn/proc-macro2; new error type produced

File: Makefile (hunk 1 above)
Comment (pjkundert) [reply Makefile:9]: No problem, removed.  The Makefiles I add are mostly documentary; I always have to figure out what incantations to run in order to build, test, etc., so I just encode them as a Makefile so I don't have to always figure it out from scratch each time I return to the project...
